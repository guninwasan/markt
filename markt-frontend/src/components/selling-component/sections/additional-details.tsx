import React from "react";
import {
  FormGroup,
  Label,
  TextArea,
  SectionHeader,
} from "../selling-component.styles";

const AdditionalDetails = ({ formData, handleChange }: any) => (
  <>
    <SectionHeader>Additional Details</SectionHeader>
    <FormGroup>
      <Label>Additional Details</Label>
      <TextArea
        name="additionalDetails"
        placeholder="Add any extra information about the product"
        value={formData.additionalDetails}
        onChange={handleChange}
      />
    </FormGroup>
  </>
);

export { AdditionalDetails };
