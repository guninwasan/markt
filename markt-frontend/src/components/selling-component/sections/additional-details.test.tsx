import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { AdditionalDetails } from "./additional-details";

describe("AdditionalDetails Component", () => {
  const mockHandleChange = jest.fn();
  const formData = {
    additionalDetails: "Some additional details",
  };

  beforeEach(() => {
    render(
      <AdditionalDetails formData={formData} handleChange={mockHandleChange} />
    );
  });

  it("should render section header", () => {
    expect(
      screen.getByTestId("additional-details-section-header")
    ).toBeInTheDocument();
  });

  it("should render label", () => {
    expect(
      screen.getByLabelText("Additional Details (Max 200 Chars.)")
    ).toBeInTheDocument();
  });

  it("should render text area with correct value", () => {
    const textArea = screen.getByPlaceholderText(
      "Add any extra information about the product"
    );
    expect(textArea).toBeInTheDocument();
    expect(textArea).toHaveValue(formData.additionalDetails);
  });

  it("should call handleChange on text area change", () => {
    const textArea = screen.getByPlaceholderText(
      "Add any extra information about the product"
    );
    fireEvent.change(textArea, { target: { value: "Updated details" } });
    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <AdditionalDetails formData={formData} handleChange={mockHandleChange} />
    );
    expect(container).toMatchSnapshot();
  });
});
