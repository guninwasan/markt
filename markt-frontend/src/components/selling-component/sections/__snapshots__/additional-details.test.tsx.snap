// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalDetails Component should match snapshot 1`] = `
<div>
  <h2
    class="sc-gtLWhw eYVYUD"
    data-testid="additional-details-section-header"
  >
    Additional Details
  </h2>
  <div
    class="sc-egkSDF bHeElY"
  >
    <label
      class="sc-dntaoT bUVQgQ"
      for="additionalDetails"
    >
      Additional Details
    </label>
    <textarea
      class="sc-kLhKbu hUWHbh"
      id="additionalDetails"
      name="additionalDetails"
      placeholder="Add any extra information about the product"
    >
      Some additional details
    </textarea>
  </div>
</div>
`;
